docs/content/concepts/file-operations.md

```diff
@@ -0,0 +1,325 @@
+
+Let's trigger the full-compaction now. Make sure you have set execution mode to `batch` 
+(add an entry `execution.runtime-mode: batch`  in `flink-conf.yaml`) and run a 
+dedicated compaction job through `flink run`:
+
+```bash
+<FLINK_HOME>/bin/flink run \
+    -c org.apache.paimon.flink.action.FlinkActions \
+    /path/to/paimon-flink-**-{{< version >}}.jar \
+    compact \
+    --warehouse <warehouse-path> \
+    --database <database-name> \ 
+    --table <table-name> \
+    [--partition <partition-name>] \
+    [--catalog-conf <paimon-catalog-conf> [--catalog-conf <paimon-catalog-conf> ...]] \
+```
+
+an example would be (suppose you're already in Flink home)
+
+```bash
+./bin/flink run \
+    -c org.apache.paimon.flink.action.FlinkActions \
+    ./lib/paimon-flink-1.17-0.5-SNAPSHOT.jar \
+    compact \
+    --path file:///tmp/paimon/default.db/T
+```
+
+All current table files will be compacted and a new snapshot, namely `snapshot-4`, is 
+made and contains the following information:
+
+```json
+{
+  "version" : 3,
+  "id" : 4,
+  "schemaId" : 0,
+  "baseManifestList" : "manifest-aa28be16-82e7-4941-8b0a-7ce1c1d0fa6d-0",
+  "deltaManifestList" : "manifest-aa28be16-82e7-4941-8b0a-7ce1c1d0fa6d-1",
+  "changelogManifestList" : null,
+  "commitUser" : "a3d951d5-aa0e-4071-a5d4-4c72a4233d48",
+  "commitIdentifier" : 9223372036854775807,
+  "commitKind" : "COMPACT",
+  "timeMillis" : 1684163217960,
+  "logOffsets" : { },
+  "totalRecordCount" : 38,
+  "deltaRecordCount" : 20,
+  "changelogRecordCount" : 0,
+  "watermark" : -9223372036854775808
+}
+```
+
+The new file layout as of snapshot-4 looks like
+{{< img src="/img/file-operations-3.png">}}
+
+Note that `manifest-4-0` contains 20 manifest entries (18 `DELETE` operations and 2 `ADD` operations) 
+1. For partition `20230503` to `20230510`, two `DELETE` operations for two data files
+2. For partition `20230501` to `20230502`, one `DELETE` operation and one `ADD` operation 
+   for the same data file.
+
+
+## Alter Table
+Execute the following statement to configure full-compaction:
+```sql
+ALTER TABLE T SET ('full-compaction.delta-commits' = '1');
+```
+
+It will create a new schema for Paimon table, namely `schema-1`, but no snapshot
+has actually used this schema yet until the next commit.
+
+This configuration will ensure that partitions are full compacted before writing
+ends, and since we haven't done any compaction yet, the next commit will produce
+two snapshots, one for data written and the other for full-compaction. We will
+not use it in this example since Flink does not support running compaction in 
+Flink SQL. However, when performing CDC ingestion with this table property, you 
+can notice two snapshots created in the next commit.
+
+{{< img src="/img/file-operations-4.png">}}
+
+
+
+
+## When Data Files are Deleted
+
+Remind that the marked data files are not truly deleted until the snapshot expires and 
+no consumer depends on the snapshot.
```

Comment:
Can we add an explanation for `Flink Streaming Write` at the end? Perhaps it would be best to have a picture described here.

There are both new data, asynchronous compaction, and asynchronous snapshot expiration in it.

Reply:
I will try it out 😂！